src/SeoMoz/SeoMoz.php

```
@@ -22,6 +31,29 @@ public function __construct($accessId, $secretKey)
     }
 
     /**
+     * Make API call using an http client
+     *
+     * @param  HttpClient $client
+     * @param  string $url
+     * @throws Exception
+     * @return \SeoMoz\SeoMozData
+     */
+    public function request(HttpClient $client, $url)
+    {
+        try 
+        {
+            $response = $client->get($this->getSeoMozUrl($url));    
+        }
+
+        catch (Exception $e)
+        {
+            throw new Exception('Unable to retrieve data: '.$e->getMessage());
```

This hides the original exception. Why do you want to catch all exceptions and throw `\Exception`?